src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java

```diff
@@ -0,0 +1,410 @@
+                            log.warn(
+                                "ARM status is failed for external object id {}. ARM error description: {} ARM error status: {}",
+                                externalObjectDirectory.getId(),
+                                armResponseUploadFileRecord.getExceptionDescription(),
+                                armResponseUploadFileRecord.getErrorStatus()
+                            );
+                            updateExternalObjectDirectory(externalObjectDirectory, armResponseProcessingFailed);
+                        }
+                        deleteResponseFiles(externalObjectDirectory, uploadFileFilenameProcessor, createRecordFilename, inputUploadFilenameProcessor);
+                    } else {
+                        log.warn("Unable to read upload file {}", uploadFileFilenameProcessor.getUploadFileFilename());
+                        updateExternalObjectDirectory(externalObjectDirectory, armResponseProcessingFailed);
+                    }
+                } else {
+                    log.warn("Failed to write upload file to temp workspace {}", uploadFileFilenameProcessor.getUploadFileFilename());
+                    updateExternalObjectDirectoryStatusAndVerificationAttempt(externalObjectDirectory, armDropZoneStatus);
+                }
+            } catch (IOException e) {
+                log.error("Unable to write upload file to temporary workspace {} - {}", uploadFileFilenameProcessor.getUploadFileFilename(), e.getMessage());
+                updateExternalObjectDirectoryStatusAndVerificationAttempt(externalObjectDirectory, armDropZoneStatus);
+            } catch (Exception e) {
+                log.error("Unable to process arm response upload file {}", uploadFileFilenameProcessor.getUploadFileFilename());
+                updateExternalObjectDirectory(externalObjectDirectory, armResponseProcessingFailed);
+            } finally {
+                cleanupTemporaryJsonFile(jsonPath);
+            }
+        } else {
+            updateExternalObjectDirectoryStatusAndVerificationAttempt(externalObjectDirectory, armDropZoneStatus);
+        }
+    }
+
+    private void cleanupTemporaryJsonFile(Path jsonPath) {
+        if (nonNull(jsonPath)) {
+            try {
+                if (!Files.deleteIfExists(jsonPath)) {
+                    log.warn("Deleting temporary file: {} failed", jsonPath.toFile().toString());
+                }
+            } catch (Exception e) {
+                log.error("Unable to delete temporary file: {}", jsonPath.toFile().toString());
+            }
+        }
+    }
+
+    private void processUploadFileDataSuccess(ArmResponseUploadFileRecord armResponseUploadFileRecord,
+                                              ExternalObjectDirectoryEntity externalObjectDirectory,
+                                              UploadFileFilenameProcessor uploadFileFilenameProcessor,
```

Comment:
is uploadFileFilenameProcessor, createRecordFilename, and inputUploadFilenameProcessor used?

Reply:
This was in there when the delete was going to happen but it was decided the deletion needed to be though out so was taken out but I forgot to update the method params. Removed them now.